src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsReconciliationService.kt

```diff
@@ -271,7 +283,7 @@ class NonAssociationsReconciliationService(
     dpsReasonEnum: NonAssociation.Reason,
   ): Boolean {
     val t = translateToRolesAndReason(reason, recipReason)
-    return t.first != role1 || t.second != role2 || ((reason == "BUL" || reason == "RIV") && t.third != dpsReasonEnum)
+    return t.first != role1 || t.second != role2 // too strict: || ((reason == "BUL" || reason == "RIV") && t.third != dpsReasonEnum)
```

Comment:
This comment makes sense for this commit but will it mean anything in the future? Maybe better to remove the comment from the code but add it to the Pull Request?

Reply:
I suspect I may need to remove the role checks too depending on what happens when this runs in preprod, so Ill revisit